core/blockchain.go

```diff
@@ -943,7 +965,12 @@ func (bc *BlockChain) WriteBlockWithState(ctx context.Context, block *types.Bloc
 	if err := bc.hc.WriteTd(block.Hash(), block.NumberU64(), externTd); err != nil {
 		return NonStatTy, err
 	}
-	if err := WriteBlock(bc.db, block); err != nil {
+	// Write other block data using a batch.
+	gbatch := bc.db.GlobalTable().NewBatch()
+	bbatch := bc.db.BodyTable().NewBatch()
+	hbatch := bc.db.HeaderTable().NewBatch()
+	rbatch := bc.db.ReceiptTable().NewBatch()
```

Comment:
This batch section was updated to start batching down below at `WriteBlockReceipts` - I had pulled it in wrong from upstream - but I'm not sure it makes a big difference either way. Maybe it depends on whether we still want to write the block even if something else goes wrong a little later.